### ISSUE TITLE
story: lpms roles

### ISSUE THREAD

**Issue by lucarioams** [windingtree/win-stays#36]
**Problem**

There is the need to have access to the `lpms-server` with different levels of rights.

**Solution**

Define roles that the user may have when connecting to the `lpms-server` and their corresponding roles / keys in `videre`.

**Comment by mfw78** [comment]
**Videre Roles**

As far as `videre` is concerned, there are the following roles:

```solc
enum Role {
    ADMIN,
    API,
    BIDDER,
    MANAGER,
    STAFF,
    WRITE
}
```

**NOTE: `WRITE` IS ACTUALLY A META ROLE MADE UP OF ALL THE OTHER ROLES THAT CAN WRITE TO THE DATA STORAGE**

To review these roles:

1. `ADMIN` - has complete root level access to all `service provider` (accommodation provider in `stays` parlance). Any account with this role can do ANYTHING and this must be protected at all costs. **WARNING: ANY KEY WITH THIS ROLE MUST NEVER BE STORED OUTSIDE OF SECURE STORAGE, AND MUST REMAIN IN THE SERVICE PROVIDER'S DIRECT CONTROL AT ALL TIMES.**
2. `API` - all data that is published to data storage (`IPFS` / `Swarm`) **MUST** be **SIGNED** by a key containing this role. In current `videre` design, this key never transacts on the chain, so no gas is required.
3. `BIDDER` - when the consumer asks for a stay, they receive replies from service providers. These replies **MUST** be signed by a key with the `bidder` role for the respective service provider. As with `API`, in the current `videre` design, this key never transacts on the chain, so no gas is required in this account.
4. `MANAGER` - a role that can do some privileged on-chain state changes (such as execute a refund). Current guarded functions by this are undefined.
5. `STAFF` - a role that can do non-critical on-chain state changes (such as check-in or check-out a guest). Current functions here are undefined.

Given the above, the `lpms-server` would have separate secp256k1 keys for `api`, `bidder`, `manager`, and `staff`, with a gas requirement only for `manager` and `staff`. These keys should be randomly generated on first initialisation, and never leave the server. As the `admin` retains complete root access, these keys can be revoked when moving `lpms-server` to different environments.

**LPMS Roles**

Server specific functionality (such as uploading data to storage, or responding to consumer queries via `waku`) is handled solely within `lpms-server`. Therefore, there would be no need to implement an `api` or `bidder` role as this would be done by internal `lpms-server` logic. There's no need to directly expose these via a REST api endpoint.

User specific functionality such as adjusting inventory, descriptions, availability, check-in etc are indeed roles that are user facing. Therefore there is the need to have a `MANAGER` and `STAFF` role. These would then be able to execute their corresponding functions on chain. The `manager` role may also have some `REST` endpoints exposed such as `/<roomType>/availability/block/2022-04-05` in order to block availability on the specified day.

Even though there is no `videre` admin access, this does not preclude the possible need for an `admin` role on the `lpms-server` that may be used to adjust things like e-mail servers used for pushing alert e-mails, prometheus metrics servers for logging `lpms-server` metrics etc.

**Comment by mfw78** [comment]
Roles have been implemented in `lpms-server` PR #59
